exercise/submission_models.py

```diff
@@ -234,6 +234,8 @@ def create_from_post(self, exercise, submitters, request):
         if 'lang' not in meta_data_dict:
             meta_data_dict['lang'] = get_language()
 
+        meta_data_dict["submission_hash"] = dict(submission_data_list).get("__submission_hash", "")
```

Comment:
Would it make sense to move the submission hash under the `__aplus__` key in the POST data? Most of the POST data should be part of the submission. The special fields could be all under the `__aplus__` key (dictionary) when the client sends the submission POST message. Now there is the `__aplus__` key and also another special field `__submission_hash` in the POST data.

The HTTP POST data could look like this under the special `__aplus__` field:
```
"__aplus__": {
    "lang": "en",
    "hash": "xyzabc",
}
```